File: 57-exercises-for-programmers/c-sharp/exercise_02.cs
// Create a program that prompts for an input string and displays output that
// shows the input string and the number of characters the string contains.

public class Exercise2 {
	string String { get; set; }

	public Exercise2() {
		PromptString();
		PrintStringAndLength();
	}

	void PromptString() {
		System.Console.Write("Enter String: ");
		String = System.Console.ReadLine();
	}

	void PrintStringAndLength() {
		System.Console.WriteLine();
		System.Console.WriteLine("===Output===");
		System.Console.WriteLine("String \"{0}\" has {1} characters.", String, String.Length);
	}
}

File: 57-exercises-for-programmers/rust/src/exercise_02.rs
// Create a program that prompts for an input string and displays output that
// shows the input string and the number of characters the string contains.

pub mod exercise_2 {
    pub fn run() {
        display_string_length(prompt_string());
    }

    fn prompt_string() -> String {
        use std::io::{stdin, stdout, Write};

        print!("What is the input string? ");
        stdout().flush().unwrap();

        let mut string = String::new();
        stdin()
            .read_line(&mut string)
            .expect("Error: failed to read input.");

        let string = string.trim();
        if string.len() > 0 {
            string.to_string()
        }
        else {
            eprintln!("Error: String is empty");
            prompt_string()
        }
    }

    fn display_string_length(string : String) {
        println!("{} has {} characters.", string, string.len());
    }
}
